include/zephyr/kernel.h

```diff
@@ -2926,13 +2928,19 @@ struct k_mutex {
 /**
  * @cond INTERNAL_HIDDEN
  */
+#ifdef CONFIG_MULTITHREADING
 #define Z_MUTEX_INITIALIZER(obj) \
 	{ \
 	.wait_q = Z_WAIT_Q_INIT(&obj.wait_q), \
 	.owner = NULL, \
 	.lock_count = 0, \
 	.owner_orig_prio = K_LOWEST_APPLICATION_THREAD_PRIO, \
 	}
+#else
+#define Z_MUTEX_INITIALIZER(obj) \
+	{ \
+	}
+#endif
```

Comment:
Something like this might be a bit cleaner 
```suggestion
 #define Z_MUTEX_INITIALIZER(obj) \
 	IF_ENABLED(CONFIG_MULTITHREADING, ({ \
 	.wait_q = Z_WAIT_Q_INIT(&obj.wait_q), \
 	.owner = NULL, \
 	.lock_count = 0, \
 	.owner_orig_prio = K_LOWEST_APPLICATION_THREAD_PRIO, \
 	}))
```

Reply:
Done!